packages/nimble-components/src/table/template.ts

```diff
@@ -36,18 +37,22 @@ export const template = html<Table>`
                     `)}
                 </div>
             </div>
-            <div class="table-viewport" role="rowgroup">
-            ${when(x => x.columns.length > 0, html<Table>`
-                ${repeat(x => x.tableData, html<TableRowState>`
-                    <${DesignSystem.tagFor(TableRow)}
-                        class="row"
-                        record-id="${x => x.id}"
-                        :dataRecord="${x => x.record}"
-                        :columns="${(_, c) => (c.parent as Table).columns}"
-                    >
-                    </${DesignSystem.tagFor(TableRow)}>
-                `)}
-            `)}
+            <div class="table-viewport" ${ref('viewport')}>
+                <div class="table-scroll" style="height: ${x => x.virtualizationHelper.allRowsHeight}px;"></div>
+                <div class="table-row-container" role="rowgroup" style="transform: ${x => (x.virtualizationHelper.rowContainerYOffset === 0 ? 'none' : `translateY(${x.virtualizationHelper.rowContainerYOffset}px)`)};">
```

Comment:
I played with the storybook 10K example in both Firefox and Chrome and found the scrolling to be pretty "chunky". The update rate doesn't feel too bad, but it very often seems to update the translation by multiples of the entire row height. This makes it feel like you can only scroll entire rows at a time, rather than scrolling parts of rows.

Any thoughts on what might be going on or whether this is the smoothest we can get?


https://user-images.githubusercontent.com/10500124/214887174-1a58138a-cdf7-4ef9-965e-eb0f11847bd1.mov

Reply:
>If you grab the scrollbar and scroll is it smooth?

It's hard to tell because moving the scrollbar 1px causes the table to move by a large number of rows. It subjectively feels a bit smoother, maybe? But if I reduce the row count to 100 and try scrolling, both scroll wheel and scroll bar seem chunky; unless I scroll really slowly it wants to scroll in large jumps.

Reply:
I have an update here to try: https://60e89457a987cf003efc0a5b-pptvjrevyu.chromatic.com/?path=/story/table--table&args=data:LargeDataSet

It doesn't change the scrolling amount, but it lowers a scroll delay on the virtualizer, so I think it feels smoother.

Reply:
Feels better on android, feels like it falls to a lower framerate instead of chunky (very technical descriptions lol)

Reply:
@jattasNI commented on Teams that it feels smoother for him too, so marking this Resolved.